probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -279,10 +279,13 @@ impl<'probe> Armv7a<'probe> {
         &mut self,
         op: impl FnOnce(&mut Self) -> Result<R, Error>,
     ) -> Result<R, Error> {
-        let was_running = !(self.state.current_state.is_halted() || self.core_halted()?);
+        let state_halted = self.state.current_state.is_halted();
+        let was_running = !(state_halted || self.core_halted()?);
```

Comment:
This is a bit inefficient, as you'll read from the device twice. Can you use `status` to update the cached halted state in one access?

Reply:
That's a good point, and an even better solution. I've updated it to do that.